core/lib/db_connection/src/instrument.rs

```diff
@@ -527,3 +527,30 @@ mod tests {
             .unwrap();
     }
 }
+
+pub trait DalContext<T> {
```

Comment:
Please move this code to before `mod tests`; it looks misleading otherwise. Also, I'd add this trait / impl to the `error` module, by analogy with `SqlxContext`.

Reply:
done